build/Common.props

```diff
@@ -45,8 +45,8 @@
     <SystemCollectionsImmutablePkgVer>1.4.0</SystemCollectionsImmutablePkgVer>
     <SystemDiagnosticSourcePkgVer>6.0.0</SystemDiagnosticSourcePkgVer>
     <SystemReflectionEmitLightweightPkgVer>4.7.0</SystemReflectionEmitLightweightPkgVer>
-    <SystemTextJsonPkgVer>4.7.0</SystemTextJsonPkgVer>
-    <SystemThreadingTasksExtensionsPkgVer>4.5.3</SystemThreadingTasksExtensionsPkgVer>
+    <SystemTextJsonPkgVer>6.0.5</SystemTextJsonPkgVer>
```

Comment:
@reyang Fixed package version. Otherwise, the build will fail. Since system.text.json requires System.Threading.Tasks.Extensions>= 4.5.4, I have also upgraded this package.

Reply:
This would touch the Jaeger Exporter dependency, by reducing the allowed version range from `[4.5.3, +inf)` to `[6.0.5, +inf)`. Normally libraries want to support wider range of version unless there is a strong reason not to do so.

I would suggest taking a different approach - redefine `<SystemTextJsonPkgVer>6.0.5</SystemTextJsonPkgVer>` in `Common.nonprod.props`.

Reply:
OK. I understand. Actually, it is [4.5.3, +inf) to [4.5.4, +inf) for the SystemThreadingTasksExtensionsPkgVer. Only one version difference in patch release. But the SystemTextJsonPkgVer reduced from [4.7.0, +inf) to [6.0.5, +inf). I didn't try other versions. It takes effort to try the versions starting from 4.7.0 and find a good one.

Could you explain further about these two files? If we keep the old version and restrict the working version in the nonprod file, would it work for the whole solution?

Reply:
> OK. I understand. Actually, it is [4.5.3, +inf) to [4.5.4, +inf) for the SystemThreadingTasksExtensionsPkgVer. Only one version difference in patch release. But the SystemTextJsonPkgVer reduced from [4.7.0, +inf) to [6.0.5, +inf). I didn't try other versions. It takes effort to try the versions starting from 4.7.0 and find a good one.

💯👍

> Could you explain further about these two files?

`Common.props` controls all projects - including all the packages shipped by this project, test cases and examples. It has a much higher bar - 1) version range should be carefully reviewed 2) the dependencies should be minimum 3) no dependencies on things that we don't trust (e.g. preview version, 3rd party libs) 4) any security issue in the downstream dependencies (direct or indirect) is treated as a security vulnerability of OpenTelemetry .NET

`Common.nonprod.props` inherits from `Common.props`, and is only used by projects that are not shipped (e.g. test cases, examples). It has a lower bar.

> If we keep the old version and restrict the working version in the nonprod file, would it work for the whole solution?

I _guess_ it'll "just work", although I haven't tried it.

Reply:
Thanks for the explanation, Reiley. It is really clear.
New commit pushed. You are right. It works. 🎉